lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -257,8 +257,14 @@ SPIRVType *LLVMToSPIRV::transType(Type *T) {
   if (T->isIntegerTy(1))
     return mapType(T, BM->addBoolType());
 
-  if (T->isIntegerTy())
-    return mapType(T, BM->addIntegerType(T->getIntegerBitWidth()));
+  if (T->isIntegerTy()) {
+    unsigned BitWidth = T->getIntegerBitWidth();
+    if (BM->getErrorLog().checkError(BitWidth > 1 && BitWidth <= 64,
```

Comment:
Does it make sense to check that bitwidth is precisely 8,16,32 or 64 bits?

Reply:
I wasn't aware that bit width restrictions are actually dictated by the SPIR-V spec itself, thanks for pointing that out!  Yes, in this case it makes sense to have an explicit check for those sizes.  I will update the patch and add a spec reference.